src/develop/pixelpipe_hb.c

```diff
@@ -2880,9 +2880,21 @@ float *dt_dev_get_raster_mask(const dt_dev_pixelpipe_t *pipe,
   float *raster_mask = NULL;
 
   GList *source_iter;
-  for(source_iter = pipe->nodes; source_iter; source_iter = g_list_next(source_iter))
+  for(source_iter = piece->pipe->nodes; source_iter; source_iter = g_list_next(source_iter))
   {
     const dt_dev_pixelpipe_iop_t *candidate = (dt_dev_pixelpipe_iop_t *)source_iter->data;
+
+    if((candidate->module == target_module)
+       || (candidate->module->iop_order >= target_module->iop_order))
+    {
+      dt_control_log
+        (_("can't take a rastermask from later in the pipeline"));
```

Comment:
Would be nice to have more information here, what about:

`module '%s' can't take raster mask from module '%s' which is later in the pixel pipe`

Or something like that.

Reply:
And adding also:

`raster mask is ignored`

as this is in fact non fatal and reordering the module back to proper order does restore the raster mask.

Reply:
done and agreed